src/components/PreviewFrame.jsx

```diff
@@ -141,6 +150,7 @@ class PreviewFrame extends React.Component {
 }
 
 PreviewFrame.propTypes = {
+  compiledProjectKey: PropTypes.number.isRequired,
```

Comment:
Don’t feel super-strongly about this, but I think we should either:

* Pass the entire `CompiledProject` record rather than doing `src` and `compiledProjectKey` piecemeal, or
* Not pass `compiledProjectKey` in here at all, and instead have the `<Preview>` be responsible for adding that information into the callback

Reply:
I see how I would go about the first option but not quite sure how to implement it the second way

Reply:
@pwjablonski so basically like this:

```jsx
  const projectFrames = compiledProjects.map(({source, timestamp: compiledProjectKey}, key) => (
    <PreviewFrame
      consoleEntries={consoleEntries}
      isActive={key === compiledProjects.keySeq().last()}
      key={timestamp}
      src={source}
      onConsoleError={(key, name, message) =>
        onConsoleError(key, name, message, compiledProjectKey)
      }
      onConsoleValue={(key, value) =>
        onConsoleValue(key, value, compiledProjectKey)
      }
      onRuntimeError={onRuntimeError}
    />
  ));
```

Reply:
Ahh cool! JsX doesn't seem to like arrow functions in the props though. I think I'll go about it the first way unless you have a preference?

Reply:
Ah, I think that might be an over-eager eslint rule… first way is good with me!